### DISCUSSION TITLE
Embed logo Markdown emails

### DISCUSSION THREAD

**Discussion by adriallongarriu** [laravel/framework#51325] [Q&A]
I want to change the logo for all my emails and embed the logo in to the email.
I pusblish the blades `php artisan vendor:publish --tag=laravel-mail`
Edit the file `resources\views\vendor\mail\html\message.blade.php`

In the docs https://laravel.com/docs/11.x/mail#inline-attachments it says it should be done this way:
```blade
 <img src="{{ $message->embed($pathToImage) }}">
```
But whit Markdown `$message` is null.

I tested with base64 but the image is not showing in gmail:
```html
 <img src="data:image/png;base64,{{ base64_encode(file_get_contents(public_path('/images/logo.png'))) }}"
    alt="Logo" loading="lazy" width="232" />
```

app\Mail\ContactMail.php
```php
/**
  * Get the message content definition.
  */
   public function content(): Content
   {
       return new Content(
           markdown: 'emails.contact',
       );
    }
```

I search and found https://github.com/laravel/framework/pull/47140 tested and the variable `$message` is present in `resources\views\emails\contact.blade.php` but not inside `message.blade.php`

How should I make the change or in what way should it be done because by making the change to a single place for all the emails that I have created but also those of Laravel like ResetPassword?

**Comment by asolopovas** [comment]
Same problem  this is how far it took me, you have to attach file to an email like "logo.png", which then must be referenced inside `
<img src="cid:{name of attachment}"/>`, but whenever you do this `<img src="cid:logo.png" alt="Logo">`  blade engine turns it to this 

```
<img src="cid:7616edb1a852ef44d96fe8006c3ccd13@symfony" alt="Logo" style="box-sizing: border-box; font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'; position: relative; border: none; max-width: 320px;">
```
 
where it should simply just live that markup untouched. But yest documentation is not working haven't found the solution yet not a single example

```php
<?php

namespace App\Mail;

use App\Models\Setting;
use Illuminate\Bus\Queueable;
use Illuminate\Mail\Mailable;
use Illuminate\Mail\Mailables\Address;
use Illuminate\Mail\Mailables\Attachment;
use Illuminate\Mail\Mailables\Content;
use Illuminate\Mail\Mailables\Envelope;
use Illuminate\Queue\SerializesModels;
use Illuminate\Support\Facades\Storage;

class OrderMail extends Mailable
{
    use Queueable, SerializesModels;

    public $orderData;
    public $orderPDF;
    public $logoCID;

    public function __construct($orderData = null, $orderPDF = null)
    {
        $this->orderData = $orderData;
        $this->orderPDF = $orderPDF;
    }

    public function envelope()
    {
        $subject = $this->orderData['status'] ?? 'Order Mail';

        return new Envelope(
            from: new Address('info@mail.com', 'From'),
            subject: $subject
        );
    }

    public function content()
    {
        return new Content(
            markdown: 'emails.order-mail',
            with: [
                'message' => $message,
            ]
        );
    }

    public function attachments()
    {
        $attachments = [];

        // Attach inline logo
        $attachments[] = Attachment::fromPath(public_path('img/logo.png'))
            ->as('logo.png')
            ->withMime('image/png');
       
        return $attachments;
    }
}
```

**Comment by asolopovas** [comment]
https://github.com/laravel/framework/pull/47140

this should have fixed it, but no one explain how to use it
